app/controllers/auth/AuthAction.scala

```diff
@@ -71,56 +72,60 @@ class AuthActionImpl @Inject()(
             }
             .asInstanceOf[Request[A]]
 
-          block(
-            AuthenticatedRequest[A](
-              Some(domain.Nino(trustedHelper.principalNino)),
-              None,
-              credentials,
-              confidenceLevel,
-              Some(UserName(Name(Some(trustedHelper.principalName), None))),
-              logins.previousLogin,
-              Some(trustedHelper),
-              profile,
-              trimmedRequest
-            )
-          )
-
-        case nino ~ Enrolments(enrolments) ~ Some(credentials) ~ confidenceLevel ~ name ~ logins ~ None ~ profile =>
-          val saEnrolment = enrolments.find(_.key == "IR-SA").flatMap { enrolment =>
-            enrolment.identifiers
-              .find(id => id.key == "UTR")
-              .map(key => SelfAssessmentEnrolment(SaUtr(key.value), SelfAssessmentStatus.fromString(enrolment.state)))
+          (
```

Comment:
You can include this in the top level match statements to make it a bit cleaner and easier to understand